test/FileService.Business.UnitTests/AddNewImageCommandTests.cs

```diff
@@ -108,15 +114,36 @@ public void SetUp()
         }
 
         [Test]
-        public void ShouldAddNewImageAndThumbImmage()
+        public void ShouldAddThumbImageWhenImageIsBig()
+        {
+            isBigImage = true;
+            var mapperHelper = new TestMapperHelper();
+
+            mapperMock
+                .Setup(x => x.Map(It.IsAny<ImageRequest>(), ImageType.Full, out isBigImage, null))
+                .Returns(mapperHelper.GetImage());
+
+            mapperMock
+                .Setup(x => x.Map(It.IsAny<ImageRequest>(), ImageType.Thumb, out isBigImage, firstDbImage.Id))
+                .Returns(mapperHelper.GetImage());
+
+            Assert.AreEqual(firstDbImage.Id, command.Execute(imageRequest));
+
+            validatorMock.Verify(v => v.Validate(It.IsAny<IValidationContext>()), Times.Once);
+            repositoryMock.Verify(r => r.AddNewImage(It.IsAny<DbImage>()), Times.Once);
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Full, out isBigImage, null), Times.Once);
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Thumb, out isBigImage, firstDbImage.Id), Times.Once);
+        }
+
+        [Test]
+        public void ShouldNotAddNewImageWhenImageIsNotBig()
         {
             Assert.AreEqual(firstDbImage.Id, command.Execute(imageRequest));
 
             validatorMock.Verify(v => v.Validate(It.IsAny<IValidationContext>()), Times.Once);
-            repositoryMock.Verify(r => r.AddNewImage(It.IsAny<DbImage>()), Times.Exactly(2));
-            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Full, null), Times.Once);
-            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Thumb, firstDbImage.Id), Times.Once);
-            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), It.IsAny<ImageType>(), It.IsAny<Guid?>()), Times.Exactly(2));
+            repositoryMock.Verify(r => r.AddNewImage(It.IsAny<DbImage>()), Times.Exactly(1));
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Full, out isBigImage, It.IsAny<Guid?>()), Times.Once);
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Thumb, out isBigImage, It.IsAny<Guid?>()), Times.Never);
         }
 
         [Test]
```

Comment:
_validatorMock
                .Setup(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
                .Returns(false)

Reply:
fixed